src/coreclr/tools/superpmi/superpmi-shim-collector/icorjitcompiler.cpp

```diff
@@ -36,6 +36,22 @@ CorJitResult interceptor_ICJC::compileMethod(ICorJitInfo*                comp,
                                              uint32_t* nativeSizeOfCode            /* OUT */
                                              )
 {
+    // See if we are filtering the collection (currently by simple substring match)
+    //
+    static char* filter = GetEnvironmentVariableWithDefaultA("SuperPMIShimFilter", "");
```

Comment:
You're still calling `GetEnvironmentVariableWithDefaultA` with every call, right?

You could call it in getJit() and store the filter as a member of the interceptor_ICJC class.